python/pyspark/worker.py

```diff
@@ -1057,6 +1059,9 @@ def mapper(_, it):
                     yield from eval(*[a[o] for o in args_kwargs_offsets])
                 if terminate is not None:
                     yield from terminate()
+            except StopIteration:
+                if terminate is not None:
+                    yield from terminate()
```

Comment:
If we are catching the `StopIteration` exception in mapper, do we still need the try ... catch block inside the `def func` and `def evaluate` below?

Reply:
Ah the exception thrown in eval will be caught by this 
```
except Exception as e:
    raise PySparkRuntimeError(
       error_class="UDTF_EXEC_ERROR",
```
I wonder if we should just move the whole try ... except ... block to this mapper function instead of checking this for every single input row.

Reply:
I tried this, but then we lose the UDTF method name (i.e. `eval` or `terminate`) in the error message. It seems better to keep that. I will leave this alone for now, let me know if you would prefer to continue with this change anyway and lose this information in the error message and we can proceed with that.